### ISSUE TITLE
Grafbase CLI Error on Laptop behind proxy: "encountered an error while downloading bun"

### ISSUE THREAD

**Issue by sbilello** [grafbase/grafbase#2046]
When I attempt to run `grafbase dev` on my laptop behind proxy, I receive the following error:

```
➜  grafbase git:(main) ✗ grafbase dev
Grafbase CLI 0.79.0

Error: encountered an error while downloading bun
````

I have already tried setting up the node registry, but this did not resolve the issue. Are there any other environment variables or configurations I need to adjust to allow Grafbase to run successfully in my corporate environment?

Additional Context

Grafbase CLI Version: 0.79.0
Operating System: MacOSX
Node.js Version: v22.6.0

**Comment by yoav-lavi** [comment]
Hey @sbilello!

While we're looking into this: You can install the correct version of Bun yourself provided it's on your path and in the same volume as the Grafbase CLI - We'll automatically hard-link it instead of attempting to download. 

The needed version can be found [here](https://github.com/grafbase/grafbase/blob/16a5d84b8b5aafd52ea88012aec9ab090394c8e7/cli/crates/server/src/bun.rs#L86) (currently 1.1.17), note that the version must be exact rather than equal or above to work. 

Please let me know if you need any more info and whether this works for you, thank you!

**Comment by yoav-lavi** [comment]
https://github.com/oven-sh/bun/releases/tag/bun-v1.1.17 This would be the correct release

**Comment by sbilello** [comment]
```
➜  grafbase git:(sergio-setup) ✗ bun
Bun is a fast JavaScript runtime, package manager, bundler, and test runner. (1.1.17+bb66bba1b)

Usage: bun <command> [...flags] [...args]

Commands:
  run       ./my-script.ts       Execute a file with Bun
            lint                 Run a package.json script
  test                           Run unit tests with Bun
  x         vite                 Execute a package binary (CLI), installing if needed (bunx)
  repl                           Start a REPL session with Bun
  exec                           Run a shell script directly with Bun

  install                        Install dependencies for a package.json (bun i)
  add       elysia               Add a dependency to package.json (bun a)
  remove    @parcel/core         Remove a dependency from package.json (bun rm)
  update    @shumai/shumai       Update outdated dependencies
  link      [<package>]          Register or link a local npm package
  unlink                         Unregister a local npm package
  patch <pkg>                     Prepare a package for patching
  pm <subcommand>                Additional package management utilities

  build     ./a.ts ./b.jsx       Bundle TypeScript & JavaScript into a single file

  init                           Start an empty Bun project from a blank template
  create    astro                Create a new project from a template (bun c)
  upgrade                        Upgrade to latest version of Bun.
  <command> --help               Print help text for command.

Learn more about Bun:            https://bun.sh/docs
Join our Discord community:      https://bun.sh/discord
➜  grafbase git:(sergio-setup) ✗ bun --version
1.1.17
➜  grafbase git:(sergio-setup) ✗ grafbase --version
Grafbase CLI 0.79.0
➜  grafbase git:(sergio-setup) ✗ ls
grafbase.config.ts node_modules       package-lock.json  package.json
➜  grafbase git:(sergio-setup) ✗ node --version
v22.6.0
➜  grafbase git:(sergio-setup) ✗ npm i

up to date, audited 4 packages in 737ms

2 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
➜  grafbase git:(sergio-setup) ✗ grafbase dev
Grafbase CLI 0.79.0

Error: encountered an error while downloading bun
```

@yoav-lavi Please check the above

**Comment by yoav-lavi** [comment]
@sbilello Are the CLI and bun in the same volume?

**Comment by yoav-lavi** [comment]
Alternatively, you could try putting `bun` directly in `/Users/YOURUSER/.grafbase/bun/` (`bun/` being a directory with the `bun` executable inside it)

**Comment by sbilello** [comment]
```
➜  bun pwd
/Users/sergio/.grafbase/bun
➜  bun ls -lath
total 0
drwxr-xr-x   3 sergio  staff    96B Aug 22 04:53 .
lrwxr-xr-x   2 sergio  admin    35B Aug 22 04:53 bun -> ../Cellar/bun@1.1.17/1.1.17/bin/bun
drwxr-xr-x  11 sergio  staff   352B Aug 21 18:37 ..
➜  bun
``` 
Please check the above @yoav-lavi

**Comment by sbilello** [comment]
```
➜  grafbase git:(sergio-setup) ✗ bun --version
1.1.17
➜  grafbase git:(sergio-setup) ✗ bun
Bun is a fast JavaScript runtime, package manager, bundler, and test runner. (1.1.17+bb66bba1b)

Usage: bun <command> [...flags] [...args]

Commands:
  run       ./my-script.ts       Execute a file with Bun
            lint                 Run a package.json script
  test                           Run unit tests with Bun
  x         next                 Execute a package binary (CLI), installing if needed (bunx)
  repl                           Start a REPL session with Bun
  exec                           Run a shell script directly with Bun

  install                        Install dependencies for a package.json (bun i)
  add       elysia               Add a dependency to package.json (bun a)
  remove    moment               Remove a dependency from package.json (bun rm)
  update    @shumai/shumai       Update outdated dependencies
  link      [<package>]          Register or link a local npm package
  unlink                         Unregister a local npm package
  patch <pkg>                     Prepare a package for patching
  pm <subcommand>                Additional package management utilities

  build     ./a.ts ./b.jsx       Bundle TypeScript & JavaScript into a single file

  init                           Start an empty Bun project from a blank template
  create    next-app             Create a new project from a template (bun c)
  upgrade                        Upgrade to latest version of Bun.
  <command> --help               Print help text for command.

Learn more about Bun:            https://bun.sh/docs
Join our Discord community:      https://bun.sh/discord
➜  grafbase git:(sergio-setup) ✗
```

**Comment by yoav-lavi** [comment]
Speficially I mean `grafbase dev`

**Comment by sbilello** [comment]
I removed the bun from the brew and I download and unpacked the binary in the same folder where I performed `npx grafbase init`. Please check the below commands. I am not sure what I am missing here. Thanks for your quick support @yoav-lavi 
```
➜  grafbase git:(sergio-setup) ✗ bun
Bun is a fast JavaScript runtime, package manager, bundler, and test runner. (1.1.17+bb66bba1b)

Usage: bun <command> [...flags] [...args]

Commands:
  run       ./my-script.ts       Execute a file with Bun
            lint                 Run a package.json script
  test                           Run unit tests with Bun
  x         vite                 Execute a package binary (CLI), installing if needed (bunx)
  repl                           Start a REPL session with Bun
  exec                           Run a shell script directly with Bun

  install                        Install dependencies for a package.json (bun i)
  add       @remix-run/dev       Add a dependency to package.json (bun a)
  remove    backbone             Remove a dependency from package.json (bun rm)
  update    @evan/duckdb         Update outdated dependencies
  link      [<package>]          Register or link a local npm package
  unlink                         Unregister a local npm package
  patch <pkg>                     Prepare a package for patching
  pm <subcommand>                Additional package management utilities

  build     ./a.ts ./b.jsx       Bundle TypeScript & JavaScript into a single file

  init                           Start an empty Bun project from a blank template
  create    vite                 Create a new project from a template (bun c)
  upgrade                        Upgrade to latest version of Bun.
  <command> --help               Print help text for command.

Learn more about Bun:            https://bun.sh/docs
Join our Discord community:      https://bun.sh/discord
➜  grafbase git:(sergio-setup) ✗ bun --version
1.1.17
➜  grafbase git:(sergio-setup) ✗ which bun
bun
➜  grafbase git:(sergio-setup) ✗ ls -lath
total 141864
drwxr-xr-x   2 nsz933  staff    64B Aug 22 09:03 bun-darwin-aarch64
drwxr-xr-x  11 nsz933  staff   352B Aug 22 09:03 .
-rw-r--r--   1 nsz933  staff    18M Aug 22 09:02 bun-darwin-aarch64.zip
-rw-r--r--   1 nsz933  staff   1.7K Aug 22 04:56 package-lock.json
drwxr-xr-x   4 nsz933  staff   128B Aug 22 04:53 .grafbase
drwxr-xr-x   6 nsz933  staff   192B Aug 21 22:04 node_modules
-rw-r--r--   1 nsz933  staff    12B Aug 21 22:04 .env
-rw-r--r--   1 nsz933  staff   280B Aug 21 22:04 package.json
drwxr-xr-x@ 69 nsz933  staff   2.2K Aug 21 20:52 ..
-rw-r--r--   1 nsz933  staff   947B Aug 21 18:57 grafbase.config.ts
-rwxr-xr-x   1 nsz933  staff    52M Jun 24 21:18 bun
➜  grafbase git:(sergio-setup) ✗ grafbase dev
Grafbase CLI 0.79.0

Error: encountered an error while downloading bun
^C
👋 See you next time!

```

**Comment by yoav-lavi** [comment]
Did you rename `bun-darwin-aarch64` to `bun` and move it to the `bun` directory? Hard to tell from the output

**Comment by yoav-lavi** [comment]
As in to the `~/.grafbase/bun` directory

**Comment by yoav-lavi** [comment]
Should end up as `~/.grafbase/bun/bun`

**Comment by sbilello** [comment]
I solved it after downloading the binary and moving it to `~/.grafbase/bun/bun`. It looks like that the symlink with brew doesn't work

**Comment by yoav-lavi** [comment]
@sbilello happy you're unblocked, we'll continue to look into the proxy issue

**Comment by yoav-lavi** [comment]
For future reference if you or anyone else runs into this before we can fix it:

```sh
curl --fail --location --progress-bar --output "bun.zip" https://github.com/oven-sh/bun/releases/download/bun-v1.1.17/bun-darwin-aarch64.zip && unzip bun.zip && mv bun-darwin-aarch64/bun ~/.grafbase/bun/bun
```

Should do it as a one liner

**Comment by sbilello** [comment]
Side question: Is there any way to get in the response the grafbase query plan?

**Comment by yoav-lavi** [comment]
That's on out roadmap for later this year, it's not supported yet though

**Comment by fbjork** [comment]
@sbilello quick heads up that Bun is no longer needed to start the Grafbase CLI as of version 0.80.0
